### DISCUSSION TITLE
port_handler=slirp4netns and docker-compose

### DISCUSSION THREAD

**Discussion by jurf** [containers/podman#13745] [Q&A] [answered]
Hi,

I have a [caddy-gen](https://github.com/wemake-services/caddy-gen) setup in a docker-compose. That means I have a Caddy instance serving as a reverse-proxy for other apps running in docker-compose.

I need one of the apps to be able to read source addresses, so I need Caddy to have have access to them to be able to pass it in a `X-Forwarded-For` header.

However, if I set `network_mode: "slirp4netns:port_handler=slirp4netns"` for the Caddy container, it stops seeing the other services. According to [this comment](https://github.com/containers/podman/discussions/12711#discussioncomment-1879950) that is because if I set it for a container, it will create its own network namespace.

Is there any way to circumvent that? I cannot set it to the pod as docker-compose does not create one.

**Comment by Luap99** [comment, marked as answer]
slirp4netns port handler is not support with networks (https://github.com/containers/podman/issues/8193).

compose uses networks by default the only way to set slirp and the settings is like you did with `network_mode: "slirp4netns:port_handler=slirp4netns"` but slirp4netns cannot communicate to other containers via ip.
The only way to do this would be port forwarding via the host ip.

**Reply by jurf** [reply]
I understand, thank you. For now I worked it around by using host networking. I hope this gets fixed soon.
